### ISSUE TITLE
Updating a lm object with "data=."

### ISSUE THREAD

**Issue by venpopov** [hadley/dplyr#2339]
An lm object cannot be update if it was specified as a part of a chain with argument `data=.`:

```{r}
library(dplyr)
data("iris")
ml1 <- iris %>% lm(Petal.Width ~ Sepal.Width, data=.)
update(ml1, .~. + Sepal.Length)
```

`Error in terms.formula(formula, data = data) : 
  'data' argument is of the wrong type`

**Comment by krlmlr** [comment]
dplyr is not involved here at all; I'm getting a different error message too. Please ask on StackOverflow and use the magrittr tag.

**Comment by kevinushey** [comment]
This is just a consequence of how `update()` performs its non-standard evaluation. `lm()` saves the call used to construct the model fit:

```
> ml <- iris %>% lm(Petal.Width ~ Sepal.Width, data = .)
> ml

Call:
lm(formula = Petal.Width ~ Sepal.Width, data = .)
```

However, the data object `.` does not exist after the expression has been evaluated, and so the call to `update()` fails as a symbol called `.` can no longer be discovered.

tl;dr: `magrittr` and base R's model fitting functions are incompatible.

**Comment by hadley** [comment]
I'd argue this is because `update()` is re-evaluating in the wrong environment
